src/CTA.WebForms/Services/TaskManagerService.cs

```diff
@@ -93,27 +95,24 @@ private void UpdateStallState()
             // then we do nothing
         }
 
-        private void TryResolveStall()
+        private async void TryResolveStall()
         {
-            // NOTE: Commented out for use later, currently
-            // configured to immediately cancel oldest task
-            // on stall
+            var currentStallOccurrence = _stallOccurrenceNumber;
+            await Task.Delay((int)_stallTimeout);
 
-            // Take the delay value as a constructor parameter?
-            // var currentStallOccurrence = _stallOccurrenceNumber;
-            // await Task.Delay((int)TaskStallTimeout.Short);
             // Ensure that stall state is still active and same occurrence
-            // if (_stalled && _stallOccurrenceNumber == currentStallOccurrence)
-            // {
-            var oldestTask = _managedTasks.Values.OrderBy(managedTask => managedTask.LastStatusChange).FirstOrDefault();
-
-            if (oldestTask != null)
+            if (_stalled && _stallOccurrenceNumber == currentStallOccurrence)
             {
-                oldestTask.CancelTask();
+                var oldestTask = _managedTasks.Values.OrderBy(managedTask => managedTask.LastStatusChange).FirstOrDefault();
+
```

Comment:
In general, calls to `TryResolveStall` should be rare, and not every call to `TryResolveStall` will even reach the `OrderBy` because we expect the stall to be resolved naturally during the stall timeout period. My concern with using a `PriorityQueue` would be that our lookups by task id would no longer be O(1) in that data structure, and those calls will be a lot more frequent. We could maintain a parallel `PriorityQueue` data structure for stall resolution, but I wouldn't want to change `_managedTasks` from being a `Dictionary`.

Reply:
Thanks for the info. I agree, keeping it as a `Dictionary` seems like the right choice.